### PR TITLE
C-Code-Generation (Please don't merge ... this is used for allowing code-reviews)

### PR DESCRIPTION
**Pull request by chrisdutz**
…) so we can do code reviews on it.

### CODE REVIEW

**Review by ottobackwards — COMMENTED** [review]
Looks good,  I think we should have the camelToSnake utility look ahead and group multiple cap letter together

**Comment by chrisdutz** [comment]
Yeah that would indeed shorten some names by 50%

**Comment by chrisdutz** [comment]
So how about if the next letter is uppercase too, treat the current as the same block.

**Comment by ottobackwards** [comment]
Yes something like that

**Comment by ottobackwards** [comment]
That looks great!

File: sandbox/plc4c/spi/include/plc4c/spi/read_buffer.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+struct plc4c_spi_read_buffer {
+
+};
+typedef struct plc4c_spi_read_buffer plc4c_spi_read_buffer;
+
+uint32_t plc4c_spi_read_get_pos(plc4c_spi_read_buffer* buf);
+
+uint32_t plc4c_spi_read_get_total_bytes(plc4c_spi_read_buffer* buf);
+
```
Comment (ottobackwards) [inline sandbox/plc4c/spi/include/plc4c/spi/read_buffer.h:34]: bytes not bits?

Comment (chrisdutz) [reply sandbox/plc4c/spi/include/plc4c/spi/read_buffer.h:34]: Well these are not methods used for the normal parsing. All of the normal read methods are bit-based

**Comment by ottobackwards** [comment]
wrt read/write buffer.  I think we'll need something for endianness and float formats

**Comment by chrisdutz** [comment]
;-) ... I am currently replicating what we have in Java and here the endianess is handled internally. It's handled in the constructor of the buffer.

File: sandbox/plc4c/spi/include/plc4c/spi/read_buffer.h (hunk 1 above)
Comment (ottobackwards) [reply sandbox/plc4c/spi/include/plc4c/spi/read_buffer.h:34]: Can you point me to an example reading bits?
Why would would you pass in num_bits to read a field of a certain size like float?  Are you saying that not only is the buffer abstracting reading types from a byte array, but is also abstracting reading types from a byte array with some packing scheme?

Maybe you should put something in the headers about what you are going after?

I think there should be methods for reading just a byte* of n bytes out as well

Comment (chrisdutz) [reply sandbox/plc4c/spi/include/plc4c/spi/read_buffer.h:34]: in this case you would have an "array" typed field of type "byte" ...

Comment (ottobackwards) [reply sandbox/plc4c/spi/include/plc4c/spi/read_buffer.h:34]: I'll wait to look at this until it is commented, so I can understand what you are doing.

**Pull request merged**